src/trace-writer.js

```diff
@@ -186,10 +202,15 @@ TraceWriter.prototype.writeSpan = function(spanData) {
 TraceWriter.prototype.queueTrace_ = function(trace) {
   var that = this;
 
+  if (!that.isActive) {
+    return;
```

Comment:
I see - I think this is another compelling case for keeping "phantom" functionality in the sense that the agent object returned by `start()` and `get()` should opaquely auto-switch between no-op and full implementation depending on the status of `projectID`'s existence.

Another solution would be to somehow modify the trace policy to never allow traces, which for our plugins right now would effectively short-circuit. I'm not sure whether that's a reasonable long-term solution, but it does prevent context propagation as well.

In my mind, the exact behavior we want is: as soon as the trace agent knows that it's impossible to obtain the project ID, do not queue traces for publishing. When it comes to preventing as much unnecessary work as possible and ease of implementation, I think changing the tracing policy is the most effective way... what do you think?